app/src/main/AndroidManifest.xml

```diff
@@ -53,7 +53,8 @@
         <service android:name="LatinIME"
                 android:label="@string/english_ime_name"
                 android:permission="android.permission.BIND_INPUT_METHOD"
-                android:directBootAware="true">
+                android:directBootAware="true"
+                android:exported="true">
```

Comment:
Adding "exported" is required in 31 for all services and activities with intent-filters. I just put exported="true" for all of them, but I can change them to false if necessary.